zkvm/docs/zkvm-spec.md

```diff
@@ -371,6 +374,42 @@ Time bounds are available in the transaction as [expressions](#expression-type)
 [`mintime`](#mintime) and [`maxtime`](#maxtime).
 
 
+### Contract ID
+
+_Contract ID_ is a 32-byte commitment to the [contract](#contract-type) ([anchor](#anchor), [payload](#contract-payload) and [predicate](#predicate)) that is also a unique identifier.
+
+Contract ID is a hash of the contract’s [output structure](#output-structure):
+
+```
+T = Transcript("ZkVM.contractid")
+T.commit("contract", output_structure)
+id = T.challenge_bytes()
```

Comment:
Does `challenge_bytes` not require a string argument like `"id"`?

If not, why is it used in some places?

Reply:
it requires, i fixed it and not pushed it yet